### ISSUE TITLE
Autoconfiguration of UserDetailsService breaks authentication

### ISSUE THREAD

**Issue by rainerfrey** [spring-projects/spring-security-oauth2-boot#66]
The autoconfiguration of the UserDetailService for the UserAuthenticationConverter from #49 and #47 leads to broken access to our `@ResourceServer` application.

The application is a Spring Boot app with the spring-security-starter and this autoconfigure library. Our concept is having all relevant claims in the JWT and use this for authentication, so that our resource server has little runtime dependencies to other systems for authentication. Spring Security does however autoconfigure an in-memory user detail service, that we simply ignored up to now.

With this change, authentication fails as this autoconfigured user detail service knows - as expected -  no users that correspond to the user names that are extracted from the JWTs.

To sum up, in our use case the assumption that the existing UserDetailService bean (that is autoconfigured by the framework and had no impact up to now) is a meaningful source of user information for OAuth authentication is wrong. I understand the intention to simplify the setup for other use cases, but without a way to opt out the update just breaks our application.

**Comment by jzheaux** [comment]
Thanks for the report, @rainerfrey, I'll take a look.

In the meantime, can you confirm that the workaround of exposing your own `JwtTokenStore` is sufficient?

**Comment by rainerfrey** [comment]
I will check the workaround. I am in Germany though, I won't have a chance to look at it before tomorrow morning at work.

**Comment by rainerfrey** [comment]
Excluding `UserDetailsServiceAutoConfiguration` works for me.

**Comment by rainerfrey** [comment]
Thank you!
